org.hl7.fhir.validation.cli/src/main/java/org/hl7/fhir/validation/cli/tasks/RePackageTask.java

```diff
@@ -56,48 +56,45 @@ public void logHelp(Logger logger) {
   public void executeTask(@Nonnull ValidationService validationService, @Nonnull ValidationEngine validationEngine, @Nonnull ValidationContext validationContext, @Nonnull String[] args) throws Exception {
     boolean json = validationContext.getFormat() != FhirFormat.XML;
     String output = validationContext.getOutput();
-    File f = ManagedFileAccess.file(output);
-    ExpansionPackageGeneratorOutputType t = ExpansionPackageGeneratorOutputType.FOLDER;
-    if (f.exists() && f.isDirectory()) {
-      t = ExpansionPackageGeneratorOutputType.FOLDER;
-    } else if (output.endsWith(".zip")) {
-      t = ExpansionPackageGeneratorOutputType.ZIP;
-    } else if (output.endsWith(".tgz")) {
-      t = ExpansionPackageGeneratorOutputType.TGZ;
-    } 
-    ExpansionPackageGeneratorScope scope = ExpansionPackageGeneratorScope.IG_ONLY;
-    int c = -1;
-    for (int i = 0; i < args.length; i++) {
-      if ("-scope".equals(args[i])) {
-        c = i;
-      }
-    }
-    if (c < args.length - 1) {
-      switch (args[c+1].toLowerCase()) {
-      case "ig" :
-        scope = ExpansionPackageGeneratorScope.IG_ONLY;
-        break;
-      case "igs" : 
-        scope = ExpansionPackageGeneratorScope.ALL_IGS;
-        break;
-      case "core" :
-        scope = ExpansionPackageGeneratorScope.EVERYTHING;
-        break;
-      default: 
-        log.warn("Unknown scope "+args[c+1]);
-      }
-    }
-    IWorkerContext ctxt = validationEngine.getContext();
-    PackageReGenerator ep = new PackageReGenerator().setContext(ctxt).setScope(scope);
-    ep.setNpmId(validationContext.getPackageName());
-    for (String s : validationContext.getIgs()) {
-      ep.addPackage(s);
+
+    PackageReGenerator packageReGenerator = new PackageReGenerator()
+      .setContext(validationEngine.getContext())
+      .setOutput(output)
+      .setOutputType(getExpansionPackageGeneratorOutputType(output))
+      .setJson(json)
+      .setModes(validationContext.getModeParams())
+      .setNpmId(validationContext.getPackageName())
+      .addPackages(validationContext.getIgs());
+
+    switch (Objects.requireNonNull(Params.getParam(args, "-scope")))
```

Comment:
I realize this is left from previous code, but it really should reference the constant in Params.